spec/services/authentication_spec.rb

```diff
@@ -0,0 +1,99 @@
+describe Authentication do
+  describe '.call' do
+    let(:user) { create(:user) }
+
+    subject { described_class.call(encode_token(payload)) }
+
+    def encode_token(payload)
+      JWT.encode(
+        payload,
+        Settings.authentication.secret,
+        Settings.authentication.algorithm
+      )
+    end
+
+    context 'with a DfE-SignIn ID and email' do
+      let(:payload) do
+        {
+          email:           user.email,
+          sign_in_user_id: user.sign_in_user_id
+        }
+      end
+
+      it { should eq user }
+    end
+
+    context 'with a DfE-SignIn ID but no email' do
```

Comment:
I'm thinking there is one more combo

`with a DfE-SignIn ID but mismatch email`

mismatch email get updated in db

Reply:
Correct, I'll add this.

Reply:
@benilovj says that's a separate piece of work, shouldn't be done as a part of this.

Reply:
oh i thought this was the mismatch work, also thinking about this dfe sign in always need to give email

Reply:
DfE sign in always gives an email I think.